packages/vite-plugin/src/main.ts

```diff
@@ -296,13 +296,14 @@ export default ({
           return;
         }
 
-        if (req.url === '/') {
+        const url = new URL(req.url, `http://${req.headers.host}`);
```

Comment:
You need to account for `req.url` possibly being `undefined`

Reply:
Oops, missed that case, added a check and let `url` be `undefined` otherwise.